internal/clients/compute/nic/nic.go

```diff
@@ -186,6 +188,8 @@ func IsNicUpToDate(cr *v1alpha1.Nic, nic sdkgo.Nic, ips []string) bool { // noli
 		return false
 	case nic.Properties.Dhcp != nil && *nic.Properties.Dhcp != cr.Spec.ForProvider.Dhcp:
 		return false
+	case nic.Properties.Dhcpv6 != nil && cr.Spec.ForProvider.DhcpV6 != nil && *nic.Properties.Dhcpv6 != *cr.Spec.ForProvider.DhcpV6:
```

Comment:
Can cr.Spec.ForProvider.DhcpV6 be nil?

Reply:
Yes, it needs in case `ipv6cidr` is empty on the LAN to which the NIC is connected. We had to have a way to not forward the default value of a boolean (e..g false) to Cloud API. This way we avoid receiving an error from Cloud API saying that we cannot use DHCPv6 if we do not have the IPV6 set on the LAN